tex/generic/pgf/systemlayer/pgfsys-dvips.def

```diff
@@ -159,28 +164,491 @@
+        % page area) available for distilling, as outlying parts get clipped
+        clippath pathbbox newpath pop pop translate pgf@refx neg pgf@top neg translate
+        % translate origin (0,0) to the reference point
+        gsave pgf@refx pgf@refy translate
+        mark /_objdef {pgfsmaskxform@\the\pgf@objectcount}
+          /BBox [clippath pathbbox newpath] /BP pdfmark grestore%
+      }%
+      \wd#2=0pt\ht#2=0pt\dp#2=0pt%
+      \box#2%
+      \pgfsys@outerinvoke{mark /EP pdfmark grestore}%
+    }}\@tempa%
+    \expandafter\xdef\csname pgfsmaskxform@#1\endcsname{\the\pgf@objectcount}%
+  }%
+}
+\def\pgfsys@usefading#1#2#3#4#5#6#7{%
+  \ifcsname pgfsmaskxform@#1\endcsname%
+  {%
+     \pgftransformreset%
+     \pgftransformcm{#2}{#3}{#4}{#5}{\pgfpoint{#6}{#7}}%
+     \csname pgfsmasktrans@#1\endcsname%
+     {%
+       \pgflowlevelsynccm%
+       \pgfsys@outerinvoke{%
+         /.begintransparencymaskgroup where {pop
+           /pgf@sys@currentcolorspace currentcolorspace def
+           /pgf@sys@currentcolor [currentcolor] cvx def
+           /pgf@sys@OverrideICC currentuserparams /OverrideICC get def % Chr. Liddell
+           <</OverrideICC true>> setuserparams                         % of Artifex
+           /DeviceGray setcolorspace
+           true <</Subtype/Luminosity>> clippath pathbbox newpath .begintransparencymaskgroup
+           gsave
+           72 Resolution div 72 VResolution div scale
+           1 DVImag div 1 DVImag div neg scale
+           mark {pgfsmaskxform@\csname pgfsmaskxform@#1\endcsname} /SP pdfmark
+           grestore
+           0 .endtransparencymask
+           <</OverrideICC pgf@sys@OverrideICC>> setuserparams % restore orig. settings
+           pgf@sys@currentcolorspace setcolorspace
+           pgf@sys@currentcolor setcolor
+           /pgfsmaskinplace true def
+         } if%
+       }%
+     }%
+     \pgftransforminvert%
+     \pgflowlevelsynccm%
+  }%
+  \else
+    \pgferror{Undefined fading '#1'}%
+  \fi%
+}
+\def\pgfsys@clipfading{%
+  \pgfsyssoftpath@invokecurrentpath%
+  \pgfsys@clipnext%
+  \pgfsys@discardpath%
+}
+
+
 %
 % Transparency groups
 %
 \def\pgfsys@transparencygroupfrombox#1{%
-  \special{ps:
-    mark /NamespacePush pdfmark % limit `{pgfxform}' name visibility
-    mark /_objdef {pgfxform} /BBox [clippath pathbbox newpath] /BP pdfmark
-    save gsave
+  \global\advance\pgf@objectcount\@ne%
+  \pgfsys@outerinvoke{%
+    mark /_objdef {pgfxform@\the\pgf@objectcount} /BBox [clippath pathbbox newpath] /BP pdfmark
+    gsave
+    /pgf@sys@foa pgffoa def /pgf@sys@soa pgfsoa def
     /.setstrokeconstantalpha where
-      {pop 1 .pgfsetfillopacityalpha 1 .pgfsetstrokeopacityalpha} if
+      {pop 1 .pgfsetfillopacityalpha 1 .pgfsetstrokeopacityalpha} if%
   }%
-  \ht#1=0pt
-  \dp#1=0pt
-  \hskip\pgf@picminx\hbox to 0pt{\box#1\hss}%
-  \special{ps:
-    grestore restore
+  \wd#1=0pt\ht#1=0pt\dp#1=0pt%
```

Comment:
Spaces *are* necessary here, otherwise TeX will just keep expanding tokens.
```suggestion
  \wd#1=0pt
  \ht#1=0pt
  \dp#1=0pt
```

Reply:
Does this apply to assignments of literal dimensions that end in `pt`? Because at other locations in the code, such assignments are terminated with `%` too, e. g. in the definition of `\pgfsys@hbox`, same file in master, lines 36--38:

````latex
\def\pgfsys@hbox#1{%
  \pgfsys@begin@idscope%
  \pgfsys@invoke{pgfs}%
    \special{ps::[end]}%
      \wd#1=0pt%
      \ht#1=0pt%
      \dp#1=0pt%
````

Reply:
Yes, it applies to all assignments.  The ones in `\pgfsys@hbox` are just wrong.